python_modules/libraries/dagster-mariadb/dagster_mariadb/partitioned_io_manager_v2.py

```diff
@@ -0,0 +1,659 @@
+                        # Check for dimension-specific column name in metadata
+                        column_name = (
+                            context.metadata.get(f"{dim_name}_column") 
+                            if context.metadata 
+                            else None
+                        ) or f"partition_{dim_name}"
+                        result[column_name] = value
+                    return result
+            except Exception:
+                pass
+            
+            # Fallback: split by separator (usually |)
+            dimension_names = [dim.name for dim in partitions_def.partitions_defs]
+            parts = str(partition_key).split("|")
+            for i, dim_name in enumerate(dimension_names):
+                if i < len(parts):
+                    column_name = (
+                        context.metadata.get(f"{dim_name}_column") 
+                        if context.metadata 
+                        else None
+                    ) or f"partition_{dim_name}"
+                    result[column_name] = parts[i]
+            
+            return result
+        
+        # Handle single dimension partitions
+        partition_column = self._get_partition_column_name(context, partitions_def)
+        
+        # For time-based partitions, store the partition key directly
+        return {partition_column: partition_key}
+
+    def _get_partition_column_name(
+        self, 
+        context: OutputContext, 
+        partitions_def
+    ) -> str:
+        """Determine the partition column name based on partition type."""
+        # Check metadata first
+        if context.metadata and "partition_column" in context.metadata:
+            return context.metadata["partition_column"]
+        
+        # Infer from partition type
+        if isinstance(partitions_def, (
+            DailyPartitionsDefinition,
+            WeeklyPartitionsDefinition,
+            MonthlyPartitionsDefinition,
+            HourlyPartitionsDefinition,
+            TimeWindowPartitionsDefinition
+        )):
+            return "partition_date"
+        
+        # Default
+        return self.default_partition_column
+
+    def _build_partition_where_clause(
+        self, 
+        partition_columns_values: Dict[str, Any]
+    ) -> str:
+        """Build WHERE clause for partition filtering."""
+        if not partition_columns_values:
+            return ""
+        
+        clauses = []
+        for column, value in partition_columns_values.items():
+            clauses.append(f"`{column}` = '{value}'")
+        
+        return " AND ".join(clauses)
+
+    def _ensure_table_exists(
+        self,
+        connection,
+        context: OutputContext,
+        df: pd.DataFrame,
+        use_native_partitioning: bool
+    ) -> None:
+        """Ensure the table exists, optionally with native partitioning."""
+        table_name = self._get_table_name(context)
+        
+        # Check if table exists
+        check_query = f"""
```

Comment:
This query is pretty fine for a general MariaDBPandasIOManager. Note that this one doesn't check for partitions. For that you might want to SELECT from `information_schema.PARTITIONS`
https://mariadb.com/docs/server/reference/system-tables/information-schema/information-schema-tables/information-schema-partitions-table